addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/Connection.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    public String requestDeviceStatusJSON() throws Exception {
+        HttpsURLConnection connection = connect(iCloudDataRequestURL);
+        String response = postRequest(connection, dataRequest);
+        connection.disconnect();
+
+        return response;
+    }
+
+    /***
+     * Sends a "find my device" request.
+     *
+     * @throws Exception
+     */
+    public void findMyDevice(String id) throws Exception {
+        String request = "{ \n \"device\": \"" + id + "\", \n \"subject\": \"Find My Device alert\" \n }";
+        HttpsURLConnection connection = connect(iCloudFindMyDeviceURL);
+        postRequest(connection, request);
+        connection.disconnect();
+    }
+
+    private void setRequestProperties(HttpsURLConnection connection, String payload)
+            throws UnsupportedEncodingException {
+        connection.setRequestProperty("Authorization", this.getBasicAuthorization());
+        connection.setRequestProperty("User-Agent", "Find iPhone/1.3 MeKit (iPad: iPhone OS/4.2.1)");
+        connection.setRequestProperty("Origin", iCloudURL);
+        connection.setRequestProperty("Content-Type", "application/json");
+        connection.setRequestProperty("charset", "utf-8");
+        connection.setRequestProperty("Accept-language", "en-us");
+        connection.setRequestProperty("Connection", "keep-alive");
+        connection.setRequestProperty("X-Apple-Find-Api-Ver", "2.0");
+        connection.setRequestProperty("X-Apple-Authscheme", "UserIdGuest");
+        connection.setRequestProperty("X-Apple-Realm-Support", "1.0");
+        connection.setRequestProperty("X-Client-Name", "iPad");
+        connection.setRequestProperty("Content-Length", Integer.toString(payload.getBytes("UTF-8").length));
+    }
+
+    private String postRequest(HttpsURLConnection connection, String payload) throws Exception {
+        connection.setDoOutput(true);
+        connection.setDoInput(true);
+        connection.setRequestMethod("POST");
+        setRequestProperties(connection, payload);
+
+        connection.getOutputStream().write(payload.getBytes("UTF-8"));
+
+        return getResponse(connection);
+    }
+
+    private HttpsURLConnection connect(URL url) throws IOException {
+        HttpsURLConnection connection;
+        connection = (HttpsURLConnection) url.openConnection();
+        return connection;
+    }
+
+    private String getResponse(HttpsURLConnection connection) {
+        try {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+
+                String inputLine;
+                StringBuffer stringBuffer = new StringBuffer();
+                while ((inputLine = reader.readLine()) != null) {
+                    stringBuffer.append(inputLine);
+                }
+
+                return stringBuffer.toString();
+            }
+        } catch (IOException e) {
+            logger.warn("Communication problem with icloud", e);
```

Comment:
I assume this can happen, if we are offline or the server is not reachable. If I am correct, you should log in debug level only and rather set the Thing status to OFFLINE-COMMUNICATION_ERROR.

Reply:
Done.